src/gameplay.ts

```diff
@@ -127,18 +138,75 @@ class GamePlay {
         gameObject.update();
         this.checkCollision(this.player, gameObject);
       }
-    }
+    } 
   }
 
   public checkCollision(player: Player, gameObject: GameObject) {
-    let d = dist(
-      player.position.x,
-      player.position.y,
-      gameObject.position.x,
-      gameObject.position.y
+  let d = dist(
+    player.position.x,
+    player.position.y,
+    gameObject.position.x,
+    gameObject.position.y
     );
-    if (d < gameObject.radius + 40) {
-      this.collisionCount++;
+      if (d < gameObject.radius + 40) {
+        this.collisionCount++;
+            
+          if (gameObject instanceof BlackHole){
+            this.player.currentHealth = this.updateHealth(player.currentHealth, gameObject);
+            gameObject.isHit = true;       
+          } 
+
+          if (gameObject instanceof Planet){
+            this.player.currentHealth = this.updateHealth(player.currentHealth, gameObject);
+            gameObject.isHit = true;
+          }
+
+          if (gameObject instanceof Meteorite){
+            this.player.currentHealth = this.updateHealth(player.currentHealth, gameObject);
+            gameObject.isHit = true;
+          }
+
+          if (gameObject instanceof SpaceDiamond){
+            this.player.currentHealth = this.updateHealth(player.currentHealth, gameObject);
+            gameObject.isHit = true;
+          }
+      } 
+        return gameObject.isHit = false;
+  }   
+
+
+  private shoot() {
+    if (keyIsPressed) {
+      if (keyCode === 32) {
+        let shot = new Shot(this.player.position.x +90, this.player.position.y+36);
+        this.shots.push(shot); 
+      }
```

Comment:
Vet inte varför den här kom med?? Måste blivit något fel när jag löste konflikter efter jag mergade in Victors branch, dessa rader ska bara tas bort va?

Reply:
Ja, det där är nog den gamla koden på shoot som Victor gjorde om och flyttade till player-klassen :)

Reply:
Stämmer! Dessa rader ska bort från GamePlay-klassen